addons/mail/static/src/models/composer/composer.js

```diff
@@ -383,6 +461,38 @@ function factory(dependencies) {
             }
         }
 
+        /**
+         * Queue a message and start to process it.
+         */
+        insertMessageToBeSent() {
+            const message = {
+                id: this.env.models['mail.message'].getNextTemporaryId(),
+                attachments: [['link', this.attachments]],
+                body: this._convertMessageToHtml(),
+                channel_ids: this.mentionedChannels.map(channel => channel.id),
+                composer: [['link', this]],
+                message_type: 'comment',
+                date: moment(),
+                partner_ids: this.recipients.map(partner => partner.id),
+                subject: (this.subjectContent) ? this.subjectContent : undefined,
+                author: [['link', this.env.messaging.currentPartner]],
+                originThread: [['insert', {
+                    id: this.thread.id,
+                    model: this.thread.model,
+                }]],
+                is_discussion: true,
+                isPendingSend: true,
+                isTemporary: true,
+                isTransient: true,
```

Comment:
- Alphabetical order
- This is not an actual message, these are just fields that are passed for insert.